app/api/api_v1/endpoints/orders.py

```diff
@@ -75,21 +77,32 @@ async def edit_order_status_for_owner(
     if order := get_order_by_id(db, id=order_id):
         if shop := get_shop_by_id(db, shop_id=order.shop_id, owner_id=current_owner.id):
             current_status = order.status.code
-            if (
-                data.status == OrderStatusForOwner.accepted
-                or data.status == OrderStatusForOwner.declined
-            ):
-                if current_status != "pending":
+            order_status = OrderStatus(initial_state=current_status)
+            if data.status == OrderStatusForOwner.accepted:
+                try:
+                    order_status.accept_order()
+                except MachineError:
+                    raise HTTPException(
+                        status_code=400, detail="Only pending order can be accepted.",
+                    )
+            elif data.status == OrderStatusForOwner.declined:
+                try:
+                    order_status.decline_order()
+                except MachineError:
```

Comment:
If you're writing this yourself, then there is no point of using FSM. Simplify it